src/Database/Query.php

```diff
@@ -37,7 +37,7 @@ class Query implements ExpressionInterface, IteratorAggregate
     /**
      * Connection instance to be used to execute this query.
      *
-     * @var \Cake\Datasource\ConnectionInterface
+     * @var \Cake\Datasource\ConnectionInterface|\Cake\Database\Connection
```

Comment:
Since I have already changed it in 3.5, this change will cause conflict on the next Friday :(
Also, `\Cake\Database\Query` should not accept `\Cake\Datasource\ConnectionInterface`. As you said, it wouldn't work if it is not an instance of `\Cake\Database\Connection`. So I simply changed `\Cake\Datasource\ConnectionInterface` to `Cake\Database\Connection` in my commit.